src/stream/src/task/barrier_manager/managed_state.rs

```diff
@@ -610,9 +672,12 @@ impl ManagedBarrierState {
     ) -> impl Future<Output = (PartialGraphId, u64)> + '_ {
         poll_fn(|cx| {
             for (partial_graph_id, graph_state) in &mut self.graph_states {
-                if let Poll::Ready(epoch) = graph_state.poll_next_completed_epoch(cx) {
+                if let Poll::Ready(barrier) = graph_state.poll_next_completed_barrier(cx) {
+                    if let Some(actors_to_stop) = barrier.all_stop_actors() {
+                        self.current_shared_context.drop_actors(actors_to_stop);
+                    }
```

Comment:
Can we also move this clean-up step into `collect -> is_finished`?

Reply:
At the beginning I implement it in this way, but later I realize that we also need to clean up the actor infos in other compute node, but the `collect -> is_finished` can only clean up the actor on this compute node. So now I changed to clean up here.